src/reference/informative/informC04_component_reference.rst

```diff
@@ -8,4 +8,189 @@
+      # In GoodFamily.cellml:               ╵ ╵
+      model: GoodFamilyModel                ╵ ╵
+        ├─ component: wife ╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴┘ ╵
+        └─ component: husband ╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴┘
+
+    .. container:: toggle
+
+      .. container:: header
+
+        See CellML syntax
+
+      .. code-block:: xml
+
+        <model name="SurbitonNeighbourhood">
+
+          <!-- Importing two families into the neighbourhood. -->
+          <import xlink:href="GoodFamily.cellml">
+            <!-- Within the family, the members are called "wife" and "husband", but after 
+                importing into the neighbourhood, these same components are now referred
+                to by the names "BarbaraGood" and "TomGood". -->
+            <component name="BarbaraGood" component_ref="wife" />
+            <component name="TomGood" component_ref="husband" />
+          </import>
+
+          <import xlink:href="LeadbetterFamily.cellml">
+            <!-- Even though the component_ref attributes here are the same as above
+                (i.e.: "wife" and "husband") they are interpreted within the context
+                of the imported model (i.e.: the "TheLeadbetterFamilyUnit" model imported
+                from the "LeadbetterFamily.cellml" file), and so refer to separate things.
+            -->
+            <component name="MargotLeadbetter" component_ref="wife" />
+            <component name="JerryLeadbetter" component_ref="husband" />
+          </import>
+
+        </model>
+
+        <!-- Inside the GoodFamily.cellml file: -->
+        <model name="TheGoodFamilyUnit">
+          <component name="husband" />
+          <component name="wife" />
+        </model>
+
+        <!-- Inside the LeadbetterFamily.cellml file: -->
+        <model name="TheLeadbetterFamilyUnit">
+          <component name="husband" />
+          <component name="wife" />
+        </model>
+
+    Now let's make things more interesting by adding an encapsulation hierarchy:
+
+    .. code::
+
+      model: SurbitonNeighbourhood
+        ├─ component: WomenInTheNeighbourhood
+        │   ├─ component: BarbaraGood <╴╴╴╴╴╴╴╴╴╴╴┐
+        │   └─ component: MargotLeadbetter <╴╴╴╴┐ ╷
+        └─ component: MenInTheNeighbourhood     ╷ ╷
+            ├─ component: TomGood <╴╴╴╴╴╴╴╴╴╴╴┐ ╷ ╷
+            └─ component: JerryLeadbetter <╴┐ ╷ ╷ ╷
+                                            ╷ ╷ ╷ ╷
+                                    imported components
+                                            ╵ ╵ ╵ ╵
+      # In LeadbetterFamily.cellml:         ╵ ╵ ╵ ╵
+      model: LeadbetterFamilyModel          ╵ ╵ ╵ ╵
+        ├─ component: husband ╴╴╴╴╴╴╴╴╴╴╴╴╴╴┘ ╵ ╵ ╵
+        └─ component: wife ╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴┼╴┘ ╵
+                                              ╵   ╵
+      # In GoodFamily.cellml:                 ╵   ╵
+      model: GoodFamilyModel                  ╵   ╵
+        ├─ component: husband ╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴┘   ╵
+        └─ component: wife ╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴╴┘
```

Comment:
In case we decide to use plain lines:

```
model: SurbitonNeighbourhood
  ├─ component: WomenInTheNeighbourhood
  │   ├─ component: BarbaraGood <───────────┐
  │   └─ component: MargotLeadbetter <────┐ │
  └─ component: MenInTheNeighbourhood     │ │
      ├─ component: TomGood <───────────┐ │ │
      └─ component: JerryLeadbetter <─┐ │ │ │
                                      │ │ │ │
                              imported components
                                      │ │ │ │
# In LeadbetterFamily.cellml:         │ │ │ │
model: LeadbetterFamilyModel          │ │ │ │
  ├─ component: husband ──────────────┘ │ │ │
  └─ component: wife ───────────────────┼─┘ │
                                        │   │
# In GoodFamily.cellml:                 │   │
model: GoodFamilyModel                  │   │
  ├─ component: husband ────────────────┘   │
  └─ component: wife ───────────────────────┘
```

Reply:
I'm going to keep the distinction between imports/not-imports for now.  The diagrams get a lot more complicated later on!